common/lib/container-definitions/src/runtime.ts

```diff
@@ -166,6 +166,9 @@ export interface IRuntimeFactory extends IProvideRuntimeFactory {
     /**
      * Instantiates a new IRuntime for the given IContainerContext to proxy to
      * This is the main entry point to the Container's business logic
+     *
+     * @param context - container context to be supplied to the runtime
+     * @param existing - whether to instantiate for the first time or from an existing context
      */
-    instantiateRuntime(context: IContainerContext): Promise<IRuntime>;
+    instantiateRuntime(context: IContainerContext, existing?: boolean): Promise<IRuntime>;
 }
```

Comment:
The stateful property has been removed as it's not actually necessary. The existing flag is optional so a new loader can load an old runtime (with no parameter in the function definition) and just pass the parameter which will end up being ignored. So considering we're not changing the function name, there are no back-compat worries here.

Reply:
I assume that's a staging change, as you would need to make it non-optional pretty soon, right?
You could avoid this PR (and next bump to consume it) by simply casting in loader and start to pass boolean here even though TSC would not be happy (with assumption that you would follow with definition change to add this boolean and eventually remove cast in loader). It's almost same number of PRs to raise (less by one) :)

Reply:
Yes, that is the plan. However, I have identified some implementations of this interface in bohemia which will break if I make the change non-backwards compatible from the start.